dev/tools/localization/bin/gen_date_localizations.dart

```diff
@@ -46,32 +46,38 @@ Future<void> main(List<String> rawArgs) async {
 
   final bool writeToFile = parseArgs(rawArgs).writeToFile;
 
-  final File dotPackagesFile = File(path.join('packages', 'flutter_localizations', '.packages'));
-  final bool dotPackagesExists = dotPackagesFile.existsSync();
+  final File packageConfigFile = File(path.join('packages', 'flutter_localizations', '.dart_tool', 'package_config.json'));
+  final bool packageConfigExists = packageConfigFile.existsSync();
 
-  if (!dotPackagesExists) {
+  if (!packageConfigExists) {
     exitWithError(
-      'File not found: ${dotPackagesFile.path}. $_kCommandName must be run '
+      'File not found: ${packageConfigFile.path}. $_kCommandName must be run '
       'after a successful "flutter update-packages".'
     );
   }
 
-  final String pathToIntl = dotPackagesFile
-    .readAsStringSync()
-    .split('\n')
-    .firstWhere(
-      (String line) => line.startsWith('intl:'),
-      orElse: () {
-        exitWithError('intl dependency not found in ${dotPackagesFile.path}');
-        return ''; // unreachable
-      },
-    )
-    .split(':')
-    .last;
-
-  final Directory dateSymbolsDirectory = Directory(path.join(pathToIntl, 'src', 'data', 'dates', 'symbols'));
+  final List<dynamic> packages = (
```

Comment:
here and elsewhere: use `Object?` instead of `dynamic`. `dynamic` turns off the type system and makes it easy to miss some bugs.

Reply:
Done!